### PR TITLE
Fix prometheus/open-metrics format for multiple metrics with same name

### LINKED ISSUE #11364: Metrics are exposed in the format that is not compliant with Prometheus spec
Getting below exception while parsing the metrics that is available in path **/metrics** . I'm using prometheus-to-cloudwatch agent to scrap the metrics from my open liberty based microprofile application and to push cloudwatch.   During this process i'm getting below exceptions 

 **second TYPE line for metric name "base_gc_total", or TYPE reported after samples.** .  It looks like issue is due to duplication of the metric names which is mentioned below

base_gc_total
base_gc_time_seconds

Attached the complete metrics that is available in /metric path of the application
[CompleteAvailableMetrics.txt](https://github.com/OpenLiberty/open-liberty/files/4349039/CompleteAvailableMetrics.txt)

### PR DESCRIPTION
**Pull request by Channyboy**
fixes #11364 
#build

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java [hunk 1]
```diff
@@ -36,30 +37,43 @@
...
+            Number gaugeValNumber = null;
+            Object gaugeValue = null;
+            try {
+                gaugeValue = ((Gauge) currentMetricMap.get(mid)).getValue();
+            } catch (IllegalStateException e) {
+                // The forwarding gauge is likely unloaded. A warning has already been emitted
+                return;
+            }
+            if (!Number.class.isInstance(gaugeValue)) {
+                Tr.event(tc, "Skipping Prometheus output for Gauge: " + name + " of type " + ((Gauge) currentMetricMap.get(mid)).getValue().getClass());
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java:57]: Will this message be printed every time `/metrics` is served?  i.e. will it be printed a lot of times in the log?

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/writer/PrometheusMetricWriter.java [hunk 2]
```diff
@@ -97,148 +94,111 @@ protected void writeMetricsAsPrometheus(StringBuilder builder, String registryNa
...
+                PrometheusBuilder.buildConcurrentGauge(builder, metricNamePrometheus, description, currentMetricMap);
+            } else if (metricMetadata.getTypeRaw().equals(MetricType.GAUGE)) {
+                PrometheusBuilder.buildGauge(builder, metricNamePrometheus, description, currentMetricMap, conversionFactor, appendUnit);
+            } else if (metricMetadata.getTypeRaw().equals(MetricType.TIMER)) {
+                PrometheusBuilder.buildTimer(builder, metricNamePrometheus, description, currentMetricMap);
+            } else if (metricMetadata.getTypeRaw().equals(MetricType.HISTOGRAM)) {
+                PrometheusBuilder.buildHistogram(builder, metricNamePrometheus, description, currentMetricMap, conversionFactor, appendUnit);
+            } else if (metricMetadata.getTypeRaw().equals(MetricType.METERED)) {
+                PrometheusBuilder.buildMeter(builder, metricNamePrometheus, description, currentMetricMap);
+            } else {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/writer/PrometheusMetricWriter.java:134]: Do we need a `if` line for `SIMPLE_TIMER`?

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java (hunk 1 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java:57]: Yes they would be emitted every time, but only if they did something wrong! Or if something went horribly wrong. For example this event is emitted if the gauge doesn't track a Number value. The other events emitted in PrometheusMetricWriter will happen if the metadata's type isn't correctly set.

**Comment by Channyboy** [comment]
Failures happened as they depend on https://github.com/OpenLiberty/open-liberty/pull/11591 to be merged first.

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
FATS passed but network issue caused timeouts in junit tests. Need to re-run
#build

**Comment by Channyboy** [comment]
#build
#run-libby-bot

@fmhwong can you review the new changes from the last two commits

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java [hunk 3]
```diff
@@ -10,16 +10,19 @@
  *******************************************************************************/
 package com.ibm.ws.microprofile.metrics.helper;
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java:11]: Update copyright year to 2020

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
Upon reviewing the history of this PR. The last commit was just to update a copyright year. The builds prior to this were successful. Will Merge when integration branch is unlocked.

**Comment by Channyboy** [comment]
The failure i encountered in the most recent build _after_ the single commit for copyright change (of which prior to I had a successful build) indicates a feature validation check issue. This PR (and more specifically the last commit) does not modify any feature files.

**Pull request merged**